### PR TITLE
Update quicksink dependency.

### PR DESCRIPTION
**Pull request by twittner**

### CODE REVIEW

File: protocols/secio/Cargo.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ libp2p-core = { version = "0.16.0", path = "../../core" }
 log = "0.4.6"
 prost = "0.6.1"
 pin-project = "0.4.6"
-quicksink = "0.1"
+quicksink = { git = "https://github.com/paritytech/quicksink", branch = "develop" }
```
Comment (twittner) [inline protocols/secio/Cargo.toml]: If this PR is approved, the new quicksink version will be published to crates.io and this entry will become `quicksink = "0.2"`.

File: transports/websocket/Cargo.toml [hunk 2]
```diff
@@ -16,7 +16,7 @@ either = "1.5.3"
 futures = "0.3.1"
 libp2p-core = { version = "0.16.0", path = "../../core" }
 log = "0.4.8"
-quicksink = "0.1"
+quicksink = { git = "https://github.com/paritytech/quicksink", branch = "develop" }
```
Comment (twittner) [inline transports/websocket/Cargo.toml]: If this PR is approved, the new quicksink version will be published to crates.io and this entry will become `quicksink = "0.2"`.

File: transports/websocket/src/framed.rs [hunk 3]
```diff
@@ -410,7 +410,7 @@ fn location_to_multiaddr<T>(location: &str) -> Result<Multiaddr, Error<T>> {
 /// The websocket connection.
 pub struct Connection<T> {
     receiver: BoxStream<'static, Result<data::Incoming, connection::Error>>,
-    sender: Pin<Box<dyn Sink<OutgoingData, Error = connection::Error> + Send>>,
+    sender: Pin<Box<dyn Sink<OutgoingData, Error = quicksink::Error<connection::Error>> + Send>>,
```
Comment (mxinden) [inline transports/websocket/src/framed.rs]: This leaks a dependency into the public interface, correct? As the users are likely not aware of what `quicksink` is, how about doing the same trick as above converting it to an `io` error?

Comment (twittner) [reply transports/websocket/src/framed.rs]: No worries, the type stays private. The only API `Connection` exposes is `Stream<Item = std::io::Result<_, _>>` and `Sink<Error = std::io::Error>` plus some methods that utilise the `Sink` impl.

**Comment by twittner** [comment]
The motivation for the `quicksink` change ― avoiding a panic when polled after an error occurred ― are somewhat defeated by `SinkExt::sink_map_err` which uses an `FnOnce` and panics on subsequent mapping attempts.

On the upside, this clarifies the somewhat generic "In most cases, if the sink encounters an error, the sink will permanently be unable to receive items." from the Sink API documentation, i.e. one should read that without the "In most cases" part if one is interested in avoiding a panic by a second error mapping.

**Comment by twittner** [comment]
As mentioned adding `quicksink::Error` does not fit well into the `Sink` design space so I close this PR.

**Pull request closed without merging**